server/database/repositories/TeamRepository.ts

```diff
@@ -1,11 +1,66 @@
 import { TeamWithoutMS } from '../../../shared/types/Tournament';
-import TeamModel from '../models/TeamModel';
+import { UserWithoutMS } from '../../../shared/types/User';
+import AppError from '../../utils/appError';
+import TeamModel, { TeamDocument, UserDocument } from '../models/TeamModel';
+
+// change _id to id and delete __v
+// This can't be done on mongoose layer
+interface UserDb extends Omit<UserWithoutMS, 'id'> {
+  _id: string,
+}
+function toUserDb(usr: UserWithoutMS): UserDb {
+  const usrDb : UserDb = { _id: usr.id, ...usr };
+  return usrDb;
+}
+function toUser(usrDb: UserDocument): UserWithoutMS {
+  // const usr : UserWithoutMS = {id: usrDb._id, ...usrDb}; <- tak nie mozna bo zworcimy wszystkie rzeczy z mongo
+  const usr : UserWithoutMS = {
+    id: usrDb._id.toString(),
+    // .. more user fields
+  };
+  return usr;
+}
```

Comment:
```suggestion
function toUser(usrDb: UserDocument): UserWithoutMS {
  // const usr : UserWithoutMS = {id: usrDb._id, ...usrDb}; <- tak nie mozna bo zworcimy wszystkie rzeczy z mongo
  return {
    id: usrDb._id.toString(),
    // .. more user fields
  };
}
```
So why can't we use .lean()? When you will get a lot of properties, changes will be crazy

Reply:
Probably won't add a lot of properties. It will not be crazy, since returned type is UserWithoutMS, TS will  not allow you to run without specifying all UserWithoutMS fields.